### PR TITLE
[ONNX] Raise Unsupported for Grid Sample with volumetric 5D input

### LINKED ISSUE #92209: ONNX Opset 16 GridSample Does Not Support 5D Volumetric Input Tensor
### 🐛 Describe the bug

[ONNX Opset 16 GridSample](https://github.com/onnx/onnx/blob/main/docs/Operators.md#GridSample) does not support 5D volumetric input tensor. However, PyTorch can still export it. The correct behavior should be preventing the GridSample export when the input tensor is 5D volumetric. The exported ONNX file is also attached.

```python
import torch


class GridSample(torch.nn.Module):

    def __init__(self):
        super().__init__()

    def forward(self, input_tensor, grid_tensor):

        output_tensor = torch.nn.functional.grid_sample(input=input_tensor,
                                                        grid=grid_tensor,
                                                        mode='bilinear',
                                                        padding_mode='zeros',
                                                        align_corners=False)

        return output_tensor


N = 4
C = 32
D_in = 64
H_in = 64
W_in = 64
D_out = 128
H_out = 128
W_out = 128

input_shape = [N, C, D_in, H_in, W_in]
grid_shape = [N, D_out, H_out, W_out, 3]
output_shape = [N, C, D_out, H_out, W_out]

input_tensor = torch.rand(*input_shape)
grid_tensor = torch.rand(*grid_shape)

grid_sample_module = GridSample()

output_tensor = grid_sample_module(input_tensor=input_tensor, grid_tensor=grid_tensor)

assert list(output_tensor.shape) == output_shape

torch.onnx.export(grid_sample_module,
                  {"input_tensor": input_tensor, "grid_tensor": grid_tensor},
                  "grid_sample.onnx",
                  verbose=False,
                  opset_version=16)
```

[grid_sample.zip](https://github.com/pytorch/pytorch/files/10418614/grid_sample.zip)

### Versions

```
PyTorch version: 1.13.0a0+936e930
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.24.1
Libc version: glibc-2.31

Python version: 3.8.10 (default, Jun 22 2022, 20:18:18)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-56-generic-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3090
Nvidia driver version: 515.86.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.7.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] functorch==1.13.0a0+936e930
[pip3] numpy==1.22.2
[pip3] pytorch-quantization==2.1.2
[pip3] torch==1.13.0a0+936e930
[pip3] torch-tensorrt==1.3.0a0
[pip3] torchtext==0.13.0a0+fae8e8c
[pip3] torchvision==0.15.0a0
[conda] Could not collect
```

### PR DESCRIPTION
**Pull request by leimao**
Fixes #92209

### CODE REVIEW

**Comment by BowenBao** [comment]
@leimao Thanks for contributing! The test file `test_onnx_opset.py` is for a slightly different purpose. Would you mind moving your test to `test_pytorch_onnx_onnxruntime.py`? You could reference https://github.com/pytorch/pytorch/blob/d41b5d7c145f3e09c7223c2b707933266241ec9b/test/onnx/test_pytorch_onnx_onnxruntime.py#L12490
to compose your test, and assert on the error like https://github.com/pytorch/pytorch/blob/d41b5d7c145f3e09c7223c2b707933266241ec9b/test/onnx/test_pytorch_onnx_onnxruntime.py#L7561

File: torch/onnx/symbolic_opset16.py [hunk 1]
```diff
@@ -51,6 +51,11 @@ def grid_sampler(
     padding_mode_enum,
     align_corners,
 ):
+    # Check the input and grid tensor rank beforehand.
+    if len(input.shape) == 5:
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset16.py:55]: `input` is a symbolic value of type `torch.Value`, so you'd need
https://github.com/pytorch/pytorch/blob/d41b5d7c145f3e09c7223c2b707933266241ec9b/torch/onnx/symbolic_opset9.py#L1021-L1022
to get its rank.

**Comment by BowenBao** [comment]
And checkout https://github.com/pytorch/pytorch/wiki/lintrunner to fix the lint errors.

File: torch/onnx/symbolic_opset16.py (hunk 1 above)
Comment (leimao) [reply torch/onnx/symbolic_opset16.py:55]: Thank you. BTW, could you please also educate me where I can find the definition of `torch._C.Value`?

**Comment by leimao** [comment]
> @leimao Thanks for contributing! The test file `test_onnx_opset.py` is for a slightly different purpose. Would you mind moving your test to `test_pytorch_onnx_onnxruntime.py`? You could reference
> 
> https://github.com/pytorch/pytorch/blob/d41b5d7c145f3e09c7223c2b707933266241ec9b/test/onnx/test_pytorch_onnx_onnxruntime.py#L12490
> 
> 
> to compose your test, and assert on the error like
> https://github.com/pytorch/pytorch/blob/d41b5d7c145f3e09c7223c2b707933266241ec9b/test/onnx/test_pytorch_onnx_onnxruntime.py#L7561

Do you think `test_pytorch_onnx_no_runtime.py` is actually a better place for this unit test, since we actually don't need ONNX Runtime in this case?

File: torch/onnx/symbolic_opset16.py (hunk 1 above)
Comment (BowenBao) [reply torch/onnx/symbolic_opset16.py]: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/OVERVIEW.md#value is a good place to start.

**Comment by BowenBao** [comment]
> Do you think `test_pytorch_onnx_no_runtime.py` is actually a better place for this unit test

It would be easier to 'enable' the test if someone adds support for this case in the future.

**Comment by leimao** [comment]
@BowenBao Now that all the unit tests have passed, could you please review, approve, and merge? Thanks.

**Review by BowenBao — APPROVED** [review]
🚀 @pytorchbot merge -g

**Pull request closed without merging**